fc/code_generation.py

```diff
@@ -59,6 +60,67 @@ def load_template(*name):
     return env.get_template(path)
 
 
+class DataInterpolation(Quantity):
+    """Represents linear interpolation on data loaded from file, for code generation."""
+    _next_id = 0  # Used to generate unique table IDs in code
+
+    # Sympy annoyingly overwrites __new__
+    def __new__(cls, name, data, index_variable, *args, **kwargs):
+        print('interp__new__', name)
+        obj = super().__new__(cls, name, real=True)
+
+        # Record a unique ID for this table
+        obj._id = cls._next_id
+        cls._next_id += 1
+
+        # Ensure we depend on the index variable in the model's graph
+        obj._args = [index_variable]
+
+        return obj
+
+    def __init__(self, name, data, index_variable, units):
+        """Create a new interpolation construct.
+
+        :param name: the data file base name
```

Comment:
I think ``name`` is only used as an identifier here, so doesn't have to equal a file name?

Reply:
It's also used as the identifier in the generated code, and the data file name is what is passed in from actions.py.

Reply:
Yeah that's what's passed in, but it's never used as a file name, right?